From c2aad2a2978cf6d9942ddc2a36a1fba4d62e6739 Mon Sep 17 00:00:00 2001
From: Ricardo Salveti <ricardo@foundries.io>
Date: Wed, 25 Sep 2019 17:08:35 -0300
Subject: [PATCH] cli/config: support default system config

Support reading configuration from system config when available. This
allows the OS to control and update a base config.

Upstream-Status: Pending

Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
Signed-off-by: Jose Quaresma <jose.quaresma@foundries.io>
---
 cli/config/config.go | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/cli/config/config.go b/cli/config/config.go
index cbb34486a6..f9d7ac15c8 100644
--- a/cli/config/config.go
+++ b/cli/config/config.go
@@ -29,6 +29,7 @@
 	ConfigFileName = "config.json"
 	configFileDir  = ".docker"
 	contextsDir    = "contexts"
+	defaultSystemConfig = "/usr/lib/docker/config.json"
 )
 
 var (
@@ -135,6 +136,15 @@ func load(configDir string) (*configfile.ConfigFile, error) {
 	filename := filepath.Join(configDir, ConfigFileName)
 	configFile := configfile.New(filename)
 
+	// LmP: Load values from system config by default
+	if _, err := os.Stat(defaultSystemConfig); err == nil {
+		file, err := os.Open(defaultSystemConfig)
+		if err == nil {
+			configFile.LoadFromReader(file)
+			file.Close()
+		}
+	}
+
 	file, err := os.Open(filename)
 	if err != nil {
 		if os.IsNotExist(err) {
-- 
2.50.1

